File: ios/PleoChallenge/Views/Add Comment/AddCommentViewManager.m
//
//  AddCommentViewManager.m
//  PleoChallenge
//
//  Created by Furkan Cengiz on 7.11.2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(AddCommentViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(merchant, NSString)
RCT_EXPORT_VIEW_PROPERTY(onTextSubmit, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onClose, RCTDirectEventBlock)
RCT_EXTERN_METHOD(
  updateFromManager:(nonnull NSNumber *)node
)
@end

File: ios/PleoChallenge/Views/Add Comment/AddCommentViewManager.swift
//
//  AddCommentViewManager.swift
//  PleoChallenge
//
//  Created by Furkan Cengiz on 7.11.2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@objc(AddCommentViewManager)
class AddCommentViewManager: RCTViewManager {
  override func view() -> UIView! {
    return AddCommentView()
  }
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func updateFromManager(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! AddCommentView
      component.onAppear()
    }
  }
}
